docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,198 @@ Example for the `__time` dimension:
+```
+
+A property of `missingValue` can be specified at query time to hint how to handle missing values. Setting `missingValue` to `""` or `null` or omitting the property will cause the dimension's original value to be used.
+ 
+A property of `uniqueRenames` specifies if optimizations can be used which assume there is no combining of multiple names into one. For example: If ABC123 always maps to SomeCompany, that can be optimized since it is a unique rename. But if both ABC123 and DEF456 BOTH map to SomeCompany, then that is NOT a unique rename.
+ 
+Namespace updates can be set through adding a child in utf8 json form to the zookeeper path at`druid.zk.paths.namespacePath`. It is required that the namespace under the ZK path be equal to the namespace in the json. The json is of the following format:
+
+#### URI namespace update
+The remapping values for each namespace can be specified by a namespace update json object as per
+```json
+{
+  "type":"uri",
+  "namespace":"some_namespace",
+  "uri": "s3://bucket/some/key/prefix/",
+  "parseSpec":{
+    "type":"csv",
+    "columns":["key","value"]
+  },
+  "updateMs":0,
+  "versionRegex": "renames-[0-9]*\\.gz"
+}
+```
+
+The `updateMs` value specifies the period in milliseconds between checks for updates. If the source of the namespace is capable of providing a timestamp, the namespace will only be updated if it has changed since the prior tick of `updateMs`. A value of 0, an absent parameter, or `null` all mean populate once and do not attempt to update. Whenever an update occurs, the updating system will look for a file with the most recent timestamp and assume that one with the most recent data.
+
+The `versionRegex` value specifies a regex to use to determine if a filename in the parent path of the uri should be considered when trying to find the latest version. Omitting this setting or setting it equal to `null` will match to all files it can find (equivalent to using `".*"`). The search occurs in the most significant "directory" of the uri.
+
+The parseSpec can be one of a number of values. Each of the examples below would rename foo to bar, baz to bat, and buck to truck. All parseSpec types assumes each input is delimited by a new line.
+
+Data on namespaces can be acquired through the `/druid/coordinator/v1/namespaces` endpoint on the coordinator. `GET` returns all known namespaces. `POST` will submit a new namespace with update information. And `DELETE` against `/druid/coordinator/v1/namespaces/{namespace}` will remove a namespace.
+
+##### csv ParseSpec
+
+|Parameter|Description|Required|Default|
+|---------|-----------|--------|-------|
+|`columns`|The list of columns in the csv file|yes|`null`|
+|`keyColumn`|The name of the column containing the key|no|The first column|
+|`valueColumn`|The name of the column containing the value|no|The second column|
+
+*example input*
+```
+bar,something,foo
+bat,something2,baz
+truck,something3,buck
+```
+
+*example parseSpec*
+```json
+"parseSpec": {
```

Comment:
I thought about this for a bit, and since it is unique to URI namespaces, which are intended to be the mechanism to extract namespaces from line delimited data blobs, I changed it to `namespaceParseSpec`